src/prefab/hydrate.ts

```diff
@@ -61,6 +61,10 @@ export default async function hydrate(
 
       for (const dep of await get_deps(node.pkg, initial_set.has(node.project))) {
         if (children.has(dep.project)) {
+          const self = graph[dep.project]
+          if (self.pkg.constraint === target.pkg.constraint) {
```

Comment:
seems like you need to implement `Range.eq`, surely `===` will only work here if the object itself is the same object?

Reply:
This seems to do exactly what I want, in testing. Replaces the build pkg with the dep so we don't require the version we want to build.